### PR TITLE
prog_run_diags: 2D hist of water vapor path versus Q2

### PR DESCRIPTION
**Pull request by oliverwm1**
Joint distribution between water vapor path and precipitation is a useful process diagnostic. This PR adds a diagnostic and plot of the 2D histogram between water vapor path and column-integrated drying (`-<Q2>`). It also adds a diagnostic of the conditional-mean `-<Q2>`.

Added public API:
- `vcm.histogram2d`

Significant internal changes:
- compute 2D histogram of `WVP` vs `-<Q2>` and 1D histogram of `WVP`
- compute conditional average of `-<Q2>` on `WVP`
- add plots of both these histograms and conditional average to process diagnostics page

- [x] Tests added

### CODE REVIEW

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py [hunk 1]
```diff
@@ -185,6 +186,48 @@ def plot_histogram(run_diags: RunDiagnostics, varname: str) -> RawHTML:
...
+        x_bin_widths = run_diags.get_variable(run, x_bin_widths_name)
+        y_bin_widths = run_diags.get_variable(run, y_bin_widths_name)
+        x = x_bin_widths[x_bin_name]
+        y = y_bin_widths[y_bin_name]
+        xedges = np.append(x.values, x.values[-1] + x_bin_widths.values[-1])
+        yedges = np.append(y.values, y.values[-1] + y_bin_widths.values[-1])
+        xcenters = x.values + 0.5 * x_bin_widths.values
+        fig, ax = plt.subplots()
+        xx, yy = np.meshgrid(xedges, yedges)
+        ax.pcolormesh(xx, yy, count.T, norm=matplotlib.colors.LogNorm())
```
[the remark this answers is not in the document]
Comment (oliverwm1) [reply workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py]: In the end, I find the figure easier to read (and explain) when just plotting the standard 2D histogram. But I do separately plot `p(Q)` and the conditional means on top of each other, so they are easy to compare between runs.

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/computed_diagnostics.py [hunk 2]
```diff
@@ -247,9 +247,9 @@ def load_diagnostics(rundirs) -> Tuple[Metadata, Diagnostics]:
         for key, ds in diags.items()
     ]
     diagnostics = [convert_index_to_datetime(ds, "time") for ds in diagnostics]
-    diagnostics = [_add_derived_diagnostics(ds) for ds in diagnostics]
     longest_run_ds = _longest_run(diagnostics)
     diagnostics.append(_get_verification_diagnostics(longest_run_ds))
+    diagnostics = [_add_derived_diagnostics(ds) for ds in diagnostics]
```
Comment (oliverwm1) [inline workflows/diagnostics/fv3net/diagnostics/prognostic_run/computed_diagnostics.py:252]: Switched the order here, which allows computing the conditional average of the joint histograms for the verif data.

**Comment by oliverwm1** [comment]
See example [here](https://storage.cloud.google.com/vcm-ml-scratch/test-prognostic-report/831cc159f1e7/process_diagnostics.html). This is based on the short test data, so the distributions are pretty noisy.

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py [hunk 3]
```diff
@@ -164,7 +165,9 @@ def plot_cubed_sphere_map(
     )
 
 
-def plot_histogram(run_diags: RunDiagnostics, varname: str) -> RawHTML:
+def plot_histogram(
+    run_diags: RunDiagnostics, varname: str, xscale="linear", yscale="linear"
```
Comment (AnnaKwa) [inline workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py:169]: Did you want to keep the original default of "log"?

**Review by AnnaKwa — APPROVED** [review]
Thanks, LGTM!

File: workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py (hunk 3 above)
Comment (oliverwm1) [reply workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py:169]: I figured a default more in keeping with matplotlib defaults was worth the breaking change. I modified the usage for the precip histogram in static_report.py

**Pull request merged**